### PR TITLE
Update to use slog, drop zap and multierror

### PR DESCRIPTION
**Pull request by sethvargo**
Multierror is still a transitive dependency because of the jvs, but zap is gone 😄 .

### CODE REVIEW

File: clients/go/pkg/audit/interceptor.go [hunk 1]
```diff
@@ -140,9 +141,11 @@ func (i *Interceptor) UnaryInterceptor(ctx context.Context, req interface{}, inf
...
 	principal, err := i.sc.RequestPrincipal(ctx)
 	if err != nil {
-		return i.handleReturnUnary(ctx, req, handler, status.Errorf(codes.FailedPrecondition,
-			"audit interceptor failed to get request principal; this is likely a result of misconfiguration of audit client (security_context): %v %v",
-			zap.Any("security_context", i.sc), zap.Error(err)))
+		logger.ErrorContext(ctx, "audit interceptor failed to get request principal",
+			"security_context", i.sc,
+			"error", err)
+		serr := status.Errorf(codes.FailedPrecondition, "audit interceptor failed to get request principal")
+		return i.handleReturnUnary(ctx, req, handler, serr)
```
Comment (sethvargo) [inline clients/go/pkg/audit/interceptor.go:148]: ⚠️ Behavior change - returned error no longer includes information. The information wasn't useful (GoString of a pointer value).

File: clients/go/pkg/audit/interceptor.go [hunk 2]
```diff
@@ -230,9 +235,11 @@ func (i *Interceptor) StreamInterceptor(srv interface{}, ss grpc.ServerStream, i
...
 	principal, err := i.sc.RequestPrincipal(ctx)
 	if err != nil {
-		return i.handleReturnStream(ctx, ss, handler, status.Errorf(codes.FailedPrecondition,
-			"audit interceptor failed to get request principal; this is likely a result of misconfiguration of audit client (security_context): %v %v",
-			zap.Any("security_context", i.sc), zap.Error(err)))
+		logger.ErrorContext(ctx, "audit interceptor failed to get request principal",
+			"security_context", i.sc,
+			"error", err)
+		serr := status.Errorf(codes.FailedPrecondition, "audit interceptor failed to get request principal")
+		return i.handleReturnStream(ctx, ss, handler, serr)
```
Comment (sethvargo) [inline clients/go/pkg/audit/interceptor.go:242]: ⚠️ Behavior change - returned error no longer includes information. The information wasn't useful (GoString of a pointer value).

File: clients/go/test/grpc-app/main.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ func main() {
 		syscall.SIGINT, syscall.SIGTERM)
 	defer done()
 
-	logger := logging.NewFromEnv("")
+	logger := logging.New(os.Stdout, logging.LevelDebug, logging.FormatJSON, true)
```
Comment (sethvargo) [inline clients/go/test/grpc-app/main.go:45]: ⚠️ Behavior change - the shell app seems to never have been configured via the environment. I enabled the lowest level of debugging by default.

File: cmd/server/main.go [hunk 4]
```diff
@@ -33,27 +33,28 @@ import (
...
 
 func main() {
-	ctx, done := signal.NotifyContext(context.Background(), syscall.SIGINT, syscall.SIGTERM)
+	ctx, done := signal.NotifyContext(context.Background(),
+		syscall.SIGINT, syscall.SIGTERM)
 	defer done()
 
-	logger := zlogger.NewFromEnv("")
-	ctx = zlogger.WithLogger(ctx, logger)
+	logger := logging.NewFromEnv("")
```
Comment (sethvargo) [inline cmd/server/main.go:45]: ❓ Since we're refactoring, should this get a prefix now? Like `LUMBERJACK_`?

File: clients/go/test/shell/main.go [hunk 5]
```diff
@@ -59,18 +59,18 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
+	logger.DebugContext(ctx, "received request", "url", r.URL)
 
 	// Get the traceID from URL.
 	traceID := r.URL.Query().Get(traceIDKey)
 	if traceID == "" {
 		http.Error(w, "missing trace_id in the request, add one by appending the URL with ?trace_id=$TRACE_ID.", http.StatusBadRequest)
 		return
 	}
-	bluemonday.StrictPolicy().Sanitize(traceID) // sanitize user-supplied trace
+	traceID = bluemonday.StrictPolicy().Sanitize(traceID) // sanitize user-supplied trace
```
Comment (sethvargo) [inline clients/go/test/shell/main.go:70]: Oops...

Comment (yolocs) [reply clients/go/test/shell/main.go:70]: 😂

File: cmd/server/main.go (hunk 4 above)
Comment (yolocs) [reply cmd/server/main.go:46]: Sure. It's a low risk change. We've been expecting this binary runs in a container so we didn't bother to add the prefix.

**Review by yolocs — APPROVED** [review]

File: cmd/server/main.go (hunk 4 above)
Comment (sethvargo) [reply cmd/server/main.go:46]: I'll leave it for now, so I don't lose the approval.

**Pull request merged**